libsql-server/src/http/user/extract.rs

```diff
@@ -26,19 +29,23 @@ impl FromRequestParts<AppState> for RequestContext {
             .with(namespace.clone(), |ns| ns.jwt_key())
             .await??;
 
-        let auth_header = parts.headers.get(hyper::header::AUTHORIZATION);
+    // todo think how to decide if the particular auth stragegy even needs a context?
```

Comment:
We had codified this into types in one of our first attempts of the auth strategy work. We may want to revisit that at some point, but it resulted in a large diff. We tied a context to a strategy by implementing a context trait.

Reply:
removing this comment. too vague. will propose a change in a separate pr